chia/rpc/wallet_rpc_api.py

```diff
@@ -662,16 +662,29 @@ async def get_transactions(self, request: Dict) -> Dict:
         assert self.service.wallet_state_manager is not None
 
         wallet_id = int(request["wallet_id"])
-        if "start" in request:
-            start = request["start"]
-        else:
-            start = 0
-        if "end" in request:
-            end = request["end"]
+
+        all = request.get("all", False)
+
+        if all:
+            transactions = await self.service.wallet_state_manager.tx_store.get_all_transactions_for_wallet(
+                wallet_id=wallet_id,
+            )
         else:
-            end = 50
+            if "start" in request:
+                start = request["start"]
+            else:
+                start = 0
+            if "end" in request:
+                end = request["end"]
+            else:
+                end = 50
```

Comment:
There seems to be a problem here:

If "start" is included, but `start > 50` (e.g. `70`), we will call `get_transactions_between(wallet_id, 70, 50)`

Reply:
Might want to add a test for some of the combinations of including "start" "end" and "all" in the query.

Reply:
Yep the rest looks good though

Reply:
Other than `all == True` overriding it entirely, this is unchanged.  I agree that defining the behavior and testing it would be good, but perhaps it doesn't need to be addressed as part of this addition?  Though, you did make me question the sensibility of adding another 'potentially contradictory' argument to this endpoint.  Maybe add a new 'get all transactions' endpoint to keep each endpoint simpler?  (noted in https://github.com/Chia-Network/chia-blockchain/pull/7979#discussion_r720714636 as well)